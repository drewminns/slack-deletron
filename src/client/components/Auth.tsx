import React from 'react';
import { connect } from 'react-redux';
import { loginUser, logoutUser, fetchUserProfile, IInitialState } from '../store';
import { IUserReponse } from '../../shared/interfaces';

interface IAuthProps {
  loginUser: Function;
  logoutUser: Function;
  fetchUserProfile: Function;
  token: string;
  profile: IUserReponse;
}

class AuthComponent extends React.Component<IAuthProps> {
  constructor(props: IAuthProps) {
    super(props);
  }

  componentDidMount() {
    const URLtoken = new URLSearchParams(location.search).get('token');
    const token = localStorage.getItem('sd-token');

    if (URLtoken) {
      this._setTokenFromURL(URLtoken);
      return;
    } else if (!token) {
      this.props.logoutUser();
      return;
    }

    this.props.loginUser(token);
    this.props.fetchUserProfile(token);
  }

  _handleLogout = () => {
    this.props.logoutUser();
  };

  _setTokenFromURL = (token: string) => {
    this.props.loginUser(token);
  };

  render() {
    if (!this.props.token) {
      return (
        <>
          <a href="/api/auth/slack">Log in</a>
        </>
      );
    }

    const { real_name } = this.props.profile;
    return (
      <>
        <button onClick={this._handleLogout}>Log Out</button>
        <p>{real_name}</p>
      </>
    );
  }
}

const mapStateToProps = ({ user: { token, profile } }: IInitialState) => {
  return { token, profile };
};

export const Auth = connect(
  mapStateToProps,
  { loginUser, logoutUser, fetchUserProfile },
)(AuthComponent);
