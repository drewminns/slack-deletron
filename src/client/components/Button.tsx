import React from 'react';

interface IButtonProps {
  content: string;
  handleClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  size?: string;
}

const classes = ['bg-blue-900', 'hover:bg-blue-700', 'text-white', 'text-sm'];

const renderClassName = (style: string): string => {
  switch (style) {
    default:
      classes.push('px-5', 'py-2');
  }
  return classes.join(' ');
};

export const Button: React.FC<IButtonProps> = ({ content, handleClick, size = 'default' }: IButtonProps) => {
  return (
    <button className={renderClassName(size)} onClick={handleClick}>
      {content}
    </button>
  );
};

Button.displayName = 'Button';
