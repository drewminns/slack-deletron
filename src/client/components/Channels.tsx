import React from 'react';
import { IChannelResponse, IIMResponse } from '../../shared/interfaces';

interface IChannelsProps {
  channels: IChannelResponse[];
  ims: IIMResponse[];
  loggedIn: boolean;
  fetchingChannels: boolean;
  setChannel: Function;
  currentChannel: IChannelResponse | {};
}

function renderOptGroup(displayGroup: { text: string; id: string }[], title: string) {
  return (
    <optgroup label={title}>
      {displayGroup.map((item, key) => (
        <option key={key} value={item.id}>
          {item.text}
        </option>
      ))}
    </optgroup>
  );
}

const renderChannelList = (
  channels: IChannelResponse[],
  ims: IIMResponse[],
  setChannel: Function,
  activeChannel: string,
) => {
  const channelList: { text: string; id: string }[] = channels.map(channel => ({
    id: channel.id,
    text: `${channel.name} - Members: ${channel.num_members}`,
  }));

  const imList: { text: string; id: string }[] = ims.map(im => ({
    id: im.id,
    text: `${im.user_name}`,
  }));

  // const displayedResults: { text: string; id: string }[] = [...channelList, ...imList];
  return (
    <>
      <label htmlFor="channel-select">Channel</label>
      <select
        id="channel-select"
        value={activeChannel}
        onChange={e => setChannel(e)}
        className="appearance-none w-full bg-gray-200 border border-gray-200 text-gray-700 py-3 px-4 pr-8 rounded leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
      >
        <option value="">All Files</option>
        {channelList.length && renderOptGroup(channelList, 'Channels')}
        {imList.length && renderOptGroup(imList, 'Private Messages')}
      </select>
    </>
  );
};

export const Channels: React.SFC<IChannelsProps> = ({
  loggedIn,
  fetchingChannels,
  channels,
  ims,
  setChannel,
  currentChannel,
}: IChannelsProps) => {
  if (!loggedIn) {
    return <p>Not Logged In</p>;
  }

  const activeChannel = (currentChannel.hasOwnProperty('id') && (currentChannel as IChannelResponse).id) || '';
  return <>{fetchingChannels ? <p>Loading</p> : renderChannelList(channels, ims, setChannel, activeChannel)}</>;
};

Channels.displayName = 'Channels Select';
