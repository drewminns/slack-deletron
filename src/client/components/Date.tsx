import React from 'react';
import { format, subDays, parseISO } from 'date-fns';

interface IButtonProps {
  endDate: string;
  startDate: string;
  handleStartDateChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  handleEndDateChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const DateSelector: React.FC<IButtonProps> = ({
  endDate,
  startDate,
  handleStartDateChange,
  handleEndDateChange,
}: IButtonProps) => {
  const today = format(new Date(), 'yyyy-MM-dd');
  return (
    <>
      <label htmlFor="startDate">From</label>
      <input
        type="date"
        id="startData"
        max={format(subDays(parseISO(startDate), 1), 'yyyy-MM-dd')}
        onChange={handleEndDateChange}
        value={endDate}
      />
      <label htmlFor="endData">To</label>
      <input type="date" id="endData" max={today} onChange={handleStartDateChange} value={startDate} />
    </>
  );
};

DateSelector.displayName = 'Date Selector';
