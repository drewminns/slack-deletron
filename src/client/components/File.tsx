import React from 'react';
import dayjs from 'dayjs';

import { Button } from './';
import { formatBytes } from '../utils';

interface IFileProps {
  title: string;
  id: string;
  created: Date;
  image: string;
  type: string;
  size: number;
  handleDelete: Function;
}

export const File: React.FC<IFileProps> = ({ title, image, id, created, handleDelete, type, size }: IFileProps) => {
  let imageSrc = image;
  if (!image) {
    imageSrc = `https://images.unsplash.com/photo-1426901555017-389235de7b0d?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjF9`;
  }
  return (
    <div className="max-w-sm bg-white border overflow-hidden">
      <div className="relative h-56 overflow-hidden">
        <img src={imageSrc} alt={title} className="w-full h-56 object-cover object-center" />
      </div>
      <div className="px-6 pt-4">
        <span className="inline-block text-sm font-semibold text-gray-700 mr-2">{formatBytes(size)}</span>
      </div>
      <div className="px-6 pt-2 pb-4">
        <div className="font-bold text-xl mb-2">
          {title} - {type}
        </div>
        <p className="text-gray-700 text-base">{dayjs(created).format('MMM DD YYYY')}</p>
      </div>
      <div className="px-6 pb-4">
        <Button handleClick={() => handleDelete(id)} content="Delete File" />
      </div>
    </div>
  );
};

File.displayName = 'File';
