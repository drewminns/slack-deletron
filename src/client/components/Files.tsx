import React from 'react';
import { connect } from 'react-redux';
import { fetchFilesList, IInitialState } from '../store';
import { IFilePayload } from '../../shared/interfaces';

interface IChannelsProps {
  fetchFilesList: Function;
  files: IFilePayload[];
  loggedIn: boolean;
}

class FilesComponent extends React.Component<IChannelsProps> {
  render() {
    if (!this.props.loggedIn) {
      return <p>Not Logged In</p>;
    }

    return (
      <>
        <h2>Files Component</h2>
        <button onClick={() => this.props.fetchFilesList()}>Get Channels</button>
        <ul>
          {this.props.files.map((file, i) => {
            return (
              <li key={i}>
                <p>
                  {file.title} - {file.id}
                </p>
                <p>{file.created}</p>
                <div>
                  <img src={file.image} alt={file.title} />
                </div>
              </li>
            );
          })}
        </ul>
      </>
    );
  }
}

const mapStateToProps = ({ files: { files }, user: { loggedIn } }: IInitialState) => {
  return { files, loggedIn };
};

export const Files = connect(
  mapStateToProps,
  { fetchFilesList },
)(FilesComponent);
