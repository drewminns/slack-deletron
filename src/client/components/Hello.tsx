import * as React from 'react';

// tslint:disable-next-line: interface-name
export interface HelloProps {
  compiler: string;
  framework: string;
}

// 'HelloProps' describes the shape of props.
// State is never set so we use the '{}' type.
export class Hello extends React.Component<HelloProps, {}> {
  // tslint:disable-next-line: member-access
  render() {
    return (
      <h1>
        Hello from {this.props.compiler} and {this.props.framework}!
      </h1>
    );
  }
}
