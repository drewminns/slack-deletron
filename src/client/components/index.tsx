export * from './Button';
export * from './File';
export * from './TestInput';
export * from './Channels';
export * from './Date';
