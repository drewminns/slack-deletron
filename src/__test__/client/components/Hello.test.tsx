import * as React from 'react';
import { shallow } from 'enzyme';
import { Hello } from '../../../client/components/Hello';

test('Hello mounts properly', () => {
  const hello = shallow(<Hello compiler="typescript" framework="React" />);

  expect(hello.find('h1').text()).toEqual('Hello from typescript and React!');
  expect(hello).toMatchSnapshot();
});
