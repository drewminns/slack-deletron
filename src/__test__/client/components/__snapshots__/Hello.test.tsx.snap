// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello mounts properly 1`] = `
<h1>
  Hello from 
  typescript
   and 
  React
  !
</h1>
`;
